rust/otap-dataflow/crates/otap/src/batch_processor.rs

```diff
@@ -75,32 +75,60 @@ const LOG_MSG_DROP_CONVERSION_FAILED: &str =
 const LOG_MSG_BATCHING_FAILED_PREFIX: &str = "OTAP batch processor: low-level batching failed for";
 const LOG_MSG_BATCHING_FAILED_SUFFIX: &str = "; dropping";
 
+/// How to size a batch; these are not always supported.
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "lowercase")]
+pub enum Sizer {
+    /// Count requests.
+    Requests,
+    /// Count items.
+    Items,
+    /// Count bytes.
+    Bytes,
+}
+
 /// Batch processor configuration.
 #[derive(Debug, Clone, Deserialize)]
 pub struct Config {
     /// Flush current batch when this count is reached, as a
     /// minimum. Measures the number of items (logs: records, traces:
     /// spans, metrics: data points). When pending data reaches this
     /// threshold a new batch will form and be sent.
-    #[serde(default = "default_send_batch_size")]
-    pub send_batch_size: Option<NonZeroUsize>,
+    #[serde(default = "default_min_size")]
```

Comment:
the doc comment for this should be updated to reflect that `Sizer` determines whether size refers to count of items or size or request..